plutus-contract/src/Plutus/Contract/Test/ContractModel/MissingLovelace.hs

```diff
@@ -0,0 +1,30 @@
+module Plutus.Contract.Test.ContractModel.MissingLovelace
+  ( calculateDelta
+  ) where
+
+import Data.List as List
+import Ledger.Ada qualified as Ada
+import Ledger.Value (Value, noAdaValue)
+import PlutusTx.Prelude qualified as P
+import Wallet.Emulator (Wallet)
+
+calculateDelta :: Value -> Value -> Value -> Wallet -> [(Wallet, [Value])] -> Value
+calculateDelta initialDelta initialValue finalValue w txOutCosts =
```

Comment:
Add some comments on the different step in order to better understand what is going on?

Reply:
Indeed, this is really hard to follow.